.github/scripts/validate_base_images/validate_base_images.py

```diff
@@ -0,0 +1,618 @@
+#!/usr/bin/env python3
+"""
+Validate base images used in Kubeflow Pipelines components and pipelines.
+
+This script discovers all components and pipelines in the components/ and pipelines/
+directories, compiles them using kfp.compiler to generate IR YAML, and extracts
+base_image values from the pipeline specifications.
+
+Usage:
+    uv run python .github/scripts/validate_base_images/validate_base_images.py
+
+    # With custom allowed prefix:
+    uv run python .github/scripts/validate_base_images/validate_base_images.py \\
+        --allowed-prefix ghcr.io/myorg/
+"""
+
+import argparse
+import importlib.util
+import os
+import sys
+import tempfile
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+import re
+
+import yaml
+from kfp import compiler
+
+
+ALLOWED_BASE_IMAGE_PREFIX = "ghcr.io/kubeflow/"
+PYTHON_IMAGE_PATTERN = re.compile(r"^python:\d+\.\d+.*$")
```

Comment:
can we read from an allowlist file, and then document this file so we can easily update it as needed, I suspect this might be a frequent occurrence - so ensure it's properly documented and easy to find

Reply:
the allow list can be a list of patterns as well

Reply:
Oh, haven't seen the latest comment. Will do that.